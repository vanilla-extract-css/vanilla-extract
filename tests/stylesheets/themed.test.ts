import {
  getStylesheet,
  startFixture,
  TestServer,
} from '@vanilla-extract-private/test-helpers';

const workerIndex = parseInt(process.env.JEST_WORKER_ID ?? '', 10);
let testCounter = 0;

const buildTypes = ['vite', 'esbuild', 'mini-css-extract'] as const;

buildTypes.forEach((buildType) => {
  describe(`themed - ${buildType}`, () => {
    let server: TestServer;

    beforeAll(async () => {
      const portRange = 100 * workerIndex;

      server = await startFixture('themed', {
        type: buildType,
        mode: 'production',
        basePort: 12000 + portRange + testCounter++,
      });
    });

    test('should create valid stylesheet', async () => {
      expect(
        await getStylesheet(server.url, server.stylesheet),
      ).toMatchSnapshot();
    });

    afterAll(async () => {
      await server.close();
    });
  });
});
