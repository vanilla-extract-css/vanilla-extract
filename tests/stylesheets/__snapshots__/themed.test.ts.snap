// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themed - esbuild should create valid stylesheet 1`] = `
".cvta170,
:root {
  --cvta171: blue;
  --cvta172: #fff;
  --cvta173: 4px;
  --cvta174: 8px;
  --cvta175: 12px;
}
.cvta176 {
  --cvta171: green;
  --cvta172: #fff;
  --cvta173: 8px;
  --cvta174: 12px;
  --cvta175: 16px;
}
.cvta177 {
  --cvta171: pink;
  --cvta172: purple;
  --cvta173: 6px;
  --cvta174: 12px;
  --cvta175: 18px;
}
@media screen and (min-width: 768px) {
  .cvta177 {
    --cvta171: purple;
    --cvta172: pink;
  }
}
._4dtfen0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: #87ceeb;
}
@font-face {
  font-family: jteyb10;
  src: local("Impact");
}
@font-face {
  font-family: MyGlobalComicSans;
  src: local("Comic Sans MS");
}
.jteyb11 {
  display: flex;
  flex-direction: column;
  gap: var(--cvta174);
  padding: var(--cvta175);
}
.jteyb12 {
  position: relative;
  z-index: 1;
}
.jteyb13 {
  background-color: var(--cvta171, "THIS FALLBACK VALUE SHOULD NEVER BE USED");
  border-radius: 9999px;
  color: var(--cvta172);
  font-family: jteyb10;
}
.cvta176 .cvta170 .jteyb11 .jteyb13 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .jteyb12:after {
  content: "I am content";
}
html .jteyb16 {
  opacity: var(--jteyb14, 0.5);
}
html .jteyb17 {
  opacity: var(--jteyb14, var(--jteyb15, 0.25));
}
@media only screen and (min-width: 500px) {
  .jteyb11 {
    border: 1px solid var(--cvta171);
  }
  .jteyb13 {
    padding: var(--cvta173);
  }
}
@media only screen and (min-width: 1000px) {
  .jteyb13 {
    padding: var(--cvta174);
  }
}
"
`;

exports[`themed - mini-css-extract should create valid stylesheet 1`] = `
".cdwe0v0,
:root {
  --cdwe0v1: blue;
  --cdwe0v2: #fff;
  --cdwe0v3: 4px;
  --cdwe0v4: 8px;
  --cdwe0v5: 12px;
}
.cdwe0v6 {
  --cdwe0v1: green;
  --cdwe0v2: #fff;
  --cdwe0v3: 8px;
  --cdwe0v4: 12px;
  --cdwe0v5: 16px;
}
.cdwe0v7 {
  --cdwe0v1: pink;
  --cdwe0v2: purple;
  --cdwe0v3: 6px;
  --cdwe0v4: 12px;
  --cdwe0v5: 18px;
}
@media screen and (min-width: 768px) {
  .cdwe0v7 {
    --cdwe0v1: purple;
    --cdwe0v2: pink;
  }
}
._5iqv0q0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: skyblue;
}
@font-face {
  font-family: blj6yb0;
  src: local("Impact");
}
@font-face {
  font-family: MyGlobalComicSans;
  src: local("Comic Sans MS");
}
.blj6yb1 {
  display: flex;
  flex-direction: column;
  gap: var(--cdwe0v4);
  padding: var(--cdwe0v5);
}
.blj6yb2 {
  position: relative;
  z-index: 1;
}
.blj6yb3 {
  background-color: var(--cdwe0v1, "THIS FALLBACK VALUE SHOULD NEVER BE USED");
  border-radius: 9999px;
  color: var(--cdwe0v2);
  font-family: blj6yb0;
}
.cdwe0v6 .cdwe0v0 .blj6yb1 .blj6yb3 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .blj6yb2:after {
  content: "I am content";
}
html .blj6yb6 {
  opacity: var(--blj6yb4, 0.5);
}
html .blj6yb7 {
  opacity: var(--blj6yb4, var(--blj6yb5, 0.25));
}
@media only screen and (min-width: 500px) {
  .blj6yb1 {
    border: 1px solid var(--cdwe0v1);
  }
  .blj6yb3 {
    padding: var(--cdwe0v3);
  }
}
@media only screen and (min-width: 1000px) {
  .blj6yb3 {
    padding: var(--cdwe0v4);
  }
}
"
`;

exports[`themed - parcel should create valid stylesheet 1`] = `
".cdwe0v0,
:root {
  --cdwe0v1: blue;
  --cdwe0v2: #fff;
  --cdwe0v3: 4px;
  --cdwe0v4: 8px;
  --cdwe0v5: 12px;
}
.cdwe0v6 {
  --cdwe0v1: green;
  --cdwe0v2: #fff;
  --cdwe0v3: 8px;
  --cdwe0v4: 12px;
  --cdwe0v5: 16px;
}
.cdwe0v7 {
  --cdwe0v1: pink;
  --cdwe0v2: purple;
  --cdwe0v3: 6px;
  --cdwe0v4: 12px;
  --cdwe0v5: 18px;
}
@media screen and (min-width: 768px) {
  .cdwe0v7 {
    --cdwe0v1: purple;
    --cdwe0v2: pink;
  }
}
@font-face {
  font-family: blj6yb0;
  src: local(Impact);
}
@font-face {
  font-family: MyGlobalComicSans;
  src: local(Comic Sans MS);
}
.blj6yb1 {
  display: flex;
  flex-direction: column;
  gap: var(--cdwe0v4);
  padding: var(--cdwe0v5);
}
.blj6yb2 {
  position: relative;
  z-index: 1;
}
.blj6yb3 {
  background-color: var(--cdwe0v1, "THIS FALLBACK VALUE SHOULD NEVER BE USED");
  border-radius: 9999px;
  color: var(--cdwe0v2);
  font-family: blj6yb0;
}
.cdwe0v6 .cdwe0v0 .blj6yb1 .blj6yb3 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .blj6yb2:after {
  content: "I am content";
}
html .blj6yb6 {
  opacity: var(--blj6yb4, 0.5);
}
html .blj6yb7 {
  opacity: var(--blj6yb4, var(--blj6yb5, 0.25));
}
@media only screen and (min-width: 500px) {
  .blj6yb1 {
    border: 1px solid var(--cdwe0v1);
  }
  .blj6yb3 {
    padding: var(--cdwe0v3);
  }
}
@media only screen and (min-width: 1000px) {
  .blj6yb3 {
    padding: var(--cdwe0v4);
  }
}
._5iqv0q0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: #87ceeb;
}
"
`;

exports[`themed - vite should create valid stylesheet 1`] = `
".cvta170,
:root {
  --cvta171: blue;
  --cvta172: #fff;
  --cvta173: 4px;
  --cvta174: 8px;
  --cvta175: 12px;
}
.cvta176 {
  --cvta171: green;
  --cvta172: #fff;
  --cvta173: 8px;
  --cvta174: 12px;
  --cvta175: 16px;
}
.cvta177 {
  --cvta171: pink;
  --cvta172: purple;
  --cvta173: 6px;
  --cvta174: 12px;
  --cvta175: 18px;
}
@media screen and (min-width: 768px) {
  .cvta177 {
    --cvta171: purple;
    --cvta172: pink;
  }
}
._4dtfen0 {
  box-shadow: 0 0 5px red;
}
body {
  background-color: #87ceeb;
}
@font-face {
  font-family: jteyb10;
  src: local("Impact");
}
@font-face {
  font-family: MyGlobalComicSans;
  src: local("Comic Sans MS");
}
.jteyb11 {
  display: flex;
  flex-direction: column;
  gap: var(--cvta174);
  padding: var(--cvta175);
}
.jteyb12 {
  position: relative;
  z-index: 1;
}
.jteyb13 {
  background-color: var(--cvta171, "THIS FALLBACK VALUE SHOULD NEVER BE USED");
  border-radius: 9999px;
  color: var(--cvta172);
  font-family: jteyb10;
}
.cvta176 .cvta170 .jteyb11 .jteyb13 {
  font-family: MyGlobalComicSans;
  outline: 5px solid red;
}
body .jteyb12:after {
  content: "I am content";
}
html .jteyb16 {
  opacity: var(--jteyb14, 0.5);
}
html .jteyb17 {
  opacity: var(--jteyb14, var(--jteyb15, 0.25));
}
@media only screen and (min-width: 500px) {
  .jteyb11 {
    border: 1px solid var(--cvta171);
  }
  .jteyb13 {
    padding: var(--cvta173);
  }
}
@media only screen and (min-width: 1000px) {
  .jteyb13 {
    padding: var(--cvta174);
  }
}
"
`;
